application/clicommands/ScanCommand.php

```diff
@@ -4,25 +4,147 @@
+     *                                  "since-last-scan" options.
+     *
+     * **Example usage:**
+     *
+     *  Scan all targets that have not been scanned before, or whose last scan is older than a given date time:
+     *     * icingacli x509 scan --job <name> --since-last-scan="3 days"
+     *
+     *  Scan only unknown targets: icingacli x509 scan --job <name> --since-last-scan=null
+     *
+     *  Scan only known targets: icingacli x509 scan --job <name> --rescan
+     *
+     *  Rescan only known targets whose last scan is older than a given date time:
+     *    * icingacli x509 scan --job <name> --since-last-scan="5 days" --rescan
+     *
+     *  Scan all known and unknown targets: icingacli x509 scan --job <name> --full
      */
     public function indexAction()
     {
-        $this->runJob($this->getJob());
+        $name = $this->params->shiftRequired('job');
+        $fullScan = (bool) $this->params->get('full', false);
+        $rescan = (bool) $this->params->get('rescan', false);
+
+        $parallel = (int) $this->Config()->get('scan', 'parallel', 256);
+        if ($parallel <= 0) {
+            throw new Exception('The \'parallel\' option must be set to at least 1');
+        }
+
+        $jobs = $this->Config('jobs');
+        if (! $jobs->hasSection($name)) {
+            throw new Exception(sprintf('Job %s not found', $name));
+        }
+
+        $jobDescription = $this->Config('jobs')->getSection($name);
+        if (! strlen($jobDescription->get('cidrs'))) {
+            throw new Exception(sprintf('The job %s does not specify any CIDRs', $name));
+        }
+
+        $sinceLastScan = $this->params->get('since-last-scan', '-1 day');
+        if ($sinceLastScan !== null && $sinceLastScan !== 'null') {
+            if ($sinceLastScan[0] !== '-') {
+                // When the user specified "2 days" as a threshold strtotime() will compute the
+                // timestamp NOW() + 2 days, but it has to be NOW() + (-2 days)
+                $sinceLastScan = "-$sinceLastScan";
+            }
+
+            $timestamp = strtotime($sinceLastScan);
+            if ($timestamp === false) {
+                throw new InvalidArgumentException(sprintf(
+                    'The specified last scan time is in an unknown format: %s',
+                    $this->params->get('since-last-scan')
+                ));
+            }
+
+            $sinceLastScan = new DateTime();
+            $sinceLastScan->setTimestamp($timestamp);
+        }
+
+        $job = (new Job($name, $jobDescription, SniHook::getAll()))
+            ->setFullScan($fullScan)
+            ->setRescan($rescan)
+            ->setParallel($parallel);
+
+        if ($sinceLastScan instanceof DateTime) {
+            $job->setLastScan($sinceLastScan);
+        } elseif ($sinceLastScan === 'null') {
+            $job->setLastScan(null);
+        }
```

Comment:
Add `else` and throw an exception.

Reply:
I don't know what exactly you mean by that. I could also set the `setLastScan()` to `null` in an else branch. I just wanted to verify this explicitly. There is no need to raise an exception here.